src/main/java/gyro/google/cloudsql/DbInsightsConfig.java

```diff
@@ -0,0 +1,136 @@
+import gyro.core.resource.Diffable;
+import gyro.core.resource.Updatable;
+import gyro.core.validation.Range;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+
+public class DbInsightsConfig extends Diffable implements Copyable<InsightsConfig> {
+
+    private Boolean queryInsightsEnabled;
+    private Integer queryPlansPerMinute;
+    private Integer queryStringLength;
+    private Boolean recordApplicationTags;
+    private Boolean recordClientAddress;
+
+    /**
+     * When set to ``true`` query insights in enabled.
+     */
+    @Required
+    @Updatable
+    public Boolean getQueryInsightsEnabled() {
+        return queryInsightsEnabled;
+    }
+
+    public void setQueryInsightsEnabled(Boolean queryInsightsEnabled) {
+        this.queryInsightsEnabled = queryInsightsEnabled;
+    }
+
+    /**
+     * The number of query execution plans captured by Insights per minute for all queries combined. Default is 5.
+     */
+    @Updatable
+    public Integer getQueryPlansPerMinute() {
+        return queryPlansPerMinute;
+    }
+
+    public void setQueryPlansPerMinute(Integer queryPlansPerMinute) {
+        this.queryPlansPerMinute = queryPlansPerMinute;
+    }
+
+    /**
+     * The maximum query length stored in bytes. Default value: 1024 bytes.
+     * Changing query length will restart the database.
+     */
+    @Range(min = 256, max = 4500)
+    @Updatable
+    public Integer getQueryStringLength() {
+        return queryStringLength;
+    }
+
+    public void setQueryStringLength(Integer queryStringLength) {
+        this.queryStringLength = queryStringLength;
+    }
+
+    /**
+     * When set to ``true``, query Insights will record application tags from query.
+     */
+    @Updatable
+    public Boolean getRecordApplicationTags() {
+        return recordApplicationTags;
+    }
+
+    public void setRecordApplicationTags(Boolean recordApplicationTags) {
+        this.recordApplicationTags = recordApplicationTags;
+    }
+
+    /**
+     * When set to ``true``, query Insights will record client address from query.
+     */
+    @Updatable
+    public Boolean getRecordClientAddress() {
+        return recordClientAddress;
+    }
+
+    public void setRecordClientAddress(Boolean recordClientAddress) {
+        this.recordClientAddress = recordClientAddress;
+    }
+
+    @Override
+    public void copyFrom(InsightsConfig model) throws Exception {
+        setQueryInsightsEnabled(model.getQueryInsightsEnabled());
+        setQueryPlansPerMinute(model.getQueryPlansPerMinute());
+        setQueryStringLength(model.getQueryStringLength());
+        setRecordApplicationTags(model.getRecordApplicationTags());
+        setRecordClientAddress(model.getRecordClientAddress());
+    }
+
+    @Override
+    public String primaryKey() {
+        return "";
+    }
+
+    public InsightsConfig toInsightsConfig() {
+        InsightsConfig config = new InsightsConfig();
+
+        if (getQueryInsightsEnabled() != null) {
```

Comment:
Given this field is a required value, this will never be a null conditional right ?

Reply:
Fixed